### PR TITLE
[ci] Add test identities for services in test namespaces

### PR DESCRIPTION
**Pull request by daniel-goldstein**
Identities in test namespaces cannot share the same underlying cloud identity if we want to identify requests with cloud access tokens. This also means the `test` account does not need to have the union of roles of the other robot accounts, but pruning of the `test` account's roles is left until after this PR merges so we can properly assess which roles are still in use by the `test` account.

### CODE REVIEW

**Comment by daniel-goldstein** [comment]
I think test-test-gsa-key is pretty awful sounding but I'm not sure how better to call a test user for test namespaces and it's consistent with the test-auth, test-batch, etc. names.

**Comment by danking** [comment]
I don't mind it especially given the consistency.

File: infra/azure/modules/batch/main.tf [hunk 1]
```diff
@@ -215,6 +247,42 @@ resource "azurerm_role_assignment" "test_test_container_contributor" {
   principal_id         = module.test_sp.principal_id
 }
 
+resource "azurerm_role_assignment" "test_registry_viewer" {
```
Comment (danking) [inline infra/azure/modules/batch/main.tf:250]: maybe call this `test_sp_registry_viewer`? `test_registry_viewer` seems too much like test is an adjective.

File: build.yaml [hunk 2]
```diff
@@ -477,11 +482,12 @@ steps:
...
+      kubectl -n {{ default_ns.name }} get -o json secret test-test-gsa-key | jq '{apiVersion, kind, type, data, metadata: {name: "test-gsa-key"}}' | kubectl -n {{ default_ns.name }} apply -f -
+      kubectl -n {{ default_ns.name }} get -o json secret test-test-dev-gsa-key | jq '{apiVersion, kind, type, data, metadata: {name: "test-dev-gsa-key"}}' | kubectl -n {{ default_ns.name }} apply -f -
+      kubectl -n {{ default_ns.name }} get -o json secret test-auth-gsa-key | jq '{apiVersion, kind, type, data, metadata: {name: "auth-gsa-key"}}' | kubectl -n {{ default_ns.name }} apply -f -
+      kubectl -n {{ default_ns.name }} get -o json secret test-batch-gsa-key | jq '{apiVersion, kind, type, data, metadata: {name: "batch-gsa-key"}}' | kubectl -n {{ default_ns.name }} apply -f -
+      kubectl -n {{ default_ns.name }} get -o json secret test-ci-gsa-key | jq '{apiVersion, kind, type, data, metadata: {name: "ci-gsa-key"}}' | kubectl -n {{ default_ns.name }} apply -f -
+      kubectl -n {{ default_ns.name }} get -o json secret test-grafana-gsa-key | jq '{apiVersion, kind, type, data, metadata: {name: "grafana-gsa-key"}}' | kubectl -n {{ default_ns.name }} apply -f -
```
Comment (danking) [inline build.yaml:490]: we could call them `testns-test`, `testns-auth`, etc.?

File: infra/azure/modules/auth/main.tf [hunk 3]
```diff
@@ -21,6 +21,27 @@ module "auth_sp" {
...
+  display_name = "${var.resource_group_name}-test-auth"
+
+  required_resource_access {
+    resource_app_id = "00000003-0000-0000-c000-000000000000"
+
+    resource_access {
+      # Application.ReadWrite.All
+      id   = "1bfefb4e-e0b5-418b-a88f-73c46d2cc8e9"
+      type = "Role"
+    }
```
Comment (danking) [inline infra/azure/modules/auth/main.tf:34]: Is the permissions/roles grant for auth? I was expecting an `azurerm_role_assignment` that granted it, I dunno, azure app creation/deletion/sp-creation.

File: infra/gcp-broad/main.tf [hunk 4]
```diff
@@ -285,10 +277,6 @@ resource "google_container_node_pool" "vdc_nonpreemptible_pool" {
       enable_integrity_monitoring = true
       enable_secure_boot          = false
     }
-
-    workload_metadata_config {
-      mode = "GKE_METADATA"
-    }
```
Comment (danking) [inline infra/gcp-broad/main.tf:291]: stacked on that other PR I presume, yeah?

File: infra/gcp-broad/main.tf [hunk 5]
```diff
@@ -490,6 +487,23 @@ resource "google_storage_bucket_iam_member" "batch_hail_query_bucket_storage_vie
   member = "serviceAccount:${module.batch_gsa_secret.email}"
 }
 
+module "test_batch_gsa_secret" {
+  source = "./gsa"
+  name = "test-batch"
+  project = var.gcp_project
+  iam_roles = [
+    "compute.instanceAdmin.v1",
```
Comment (danking) [inline infra/gcp-broad/main.tf:495]: totally unrelated to this PR, but at some point we should scope this to the test namespaced VMs.

**Review by danking — CHANGES_REQUESTED** [review]
We have to apply the TF before we can test this right because the secrets are missing?

File: infra/azure/modules/auth/main.tf (hunk 3 above)
Comment (daniel-goldstein) [reply infra/azure/modules/auth/main.tf:34]: Ya there's a different way of assigning permissions within AAD vs permissions on other resources in Azure. This was the only way I could find at the time to create/read apps and service principals.

File: build.yaml (hunk 2 above)
Comment (daniel-goldstein) [reply build.yaml:490]: I like this better.

File: infra/gcp-broad/main.tf (hunk 4 above)
Comment (daniel-goldstein) [reply infra/gcp-broad/main.tf:291]: yup

**Comment by daniel-goldstein** [comment]
> We have to apply the TF before we can test this right because the secrets are missing?

I already applied this terraform since I didn't see another practical way to go about it and it doesn't mutate any existing resources.

File: infra/azure/modules/auth/main.tf [hunk 6]
```diff
@@ -21,6 +21,27 @@ module "auth_sp" {
...
+  }
+}
+module "test_auth_sp" {
```
Comment (danking) [inline infra/azure/modules/auth/main.tf:37]:
```suggestion
module "testns_auth_sp" {
```
yeah?

**Review by danking — CHANGES_REQUESTED** [review]
@-me when you fix and I'll approve

**Comment by daniel-goldstein** [comment]
@danking fixed

**Review by danking — APPROVED** [review]

**Comment by danking** [comment]
You're gonna have a PR bonanza today.

**Pull request merged**